shaping/wrapping.go

```diff
@@ -362,6 +362,97 @@ func (r *runMapper) mapRun(runIdx int, run Output) {
 	}
 }
 
+// RunIterator defines a type that can incrementally provide shaped text.
+type RunIterator interface {
+	// Next returns the next logical run index, shaped run, and true if there are more runs,
+	// then it advances the iterator to the next element.
+	// Otherwise it returns an undefined index, an empty Output, and false if there are no more outputs.
+	Next() (index int, run Output, isValid bool)
+	// Peek returns the same thing Next() would, but does not advance the iterator (so the
+	// next call to Next() will return the same thing).
+	Peek() (index int, run Output, isValid bool)
+	// IsFinal returns whether the Output returned by a subsequent call to Next() or Peek()
+	// is the final output for this iterator.
+	IsFinal() bool
```

Comment:
I do not understand the meaning of this method.

At first, I thought, it would be implemented for a slice as `index == len(runs) -1` , but I then read it was `len(runs) == 1` 
It don't understand this test : how is a slice of one run a special case ?

Reply:
I was able to eliminate `IsFinal` from the interface entirely, though at a slight performance cost since we have to perform several operations to do the equivalent with Save/Next/Peek/Restore.